Track 2 SDK changes for DNS resolver public preview

arpit-gagneja opened the pull request:
Track 2 SDK changes for DNS resolver public preview

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

argagnej committed:
Track 2 SDK changes for DNS resolver

muwaqar commented on sdk/dnsresolver/Azure.ResourceManager.DnsResolver/src/Azure.ResourceManager.DnsResolver.csproj:
```diff
@@ -0,0 +1,8 @@
+  <PropertyGroup>
+    <Version>1.0.0-beta.1</Version>
+    <PackageId>Azure.ResourceManager.DnsResolver</PackageId>
+    <Description>Azure Resource Manager client SDK for Azure resource provider Microsoft.dnsresolver</Description>
```
fix this

arpit-gagneja replied:
Corrected the description.

ArthurMa1978 commented on sdk/dnsresolver/Azure.ResourceManager.DnsResolver/CHANGELOG.md:
```diff
@@ -0,0 +1,7 @@
+# Release History
+
+## 1.0.0-preview.1 (Unreleased)
```
change this to 1.0.0-beta.1

ArthurMa1978 commented on sdk/dnsresolver/Azure.ResourceManager.DnsResolver/README.md:
````diff
@@ -0,0 +1,77 @@
+Install the Azure dnsresolver management library for .NET with [NuGet](https://www.nuget.org/):
+
+```PowerShell
+Install-Package Azure.ResourceManager.DnsResolver -Version 1.0.0-preview.1 
````
change this to 1.0.0-beta.1

Yao725 commented on sdk/dnsresolver/Azure.ResourceManager.DnsResolver/Directory.Build.props:
```diff
@@ -0,0 +1,6 @@
+﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Can we remove this file?

arpit-gagneja replied:
Removed.

Yao725 commented on sdk/dnsresolver/Azure.ResourceManager.DnsResolver/README.md:
```diff
@@ -0,0 +1,77 @@
+# Azure dnsresolver Management client library for .NET
```
should be `DnsResolver`.

argagnej committed:
Addressed code review suggestions

arpit-gagneja commented:
/azp run prepare-pipelines

arpit-gagneja commented:
@ArthurMa1978 @Yao725: I have addressed the comments. Also, I tried to run the pipelines but got an automated reply above saying that I don't have the privilege. Can you please run for me?

Yao725 commented:
/azp run prepare-pipelines

arpit-gagneja commented:
/azp run net - dnsresolver - ci

argagnej committed:
Added tests for DNS resolver

Yao725 commented:
/azp run net - dnsresolver - ci

argagnej committed:
Reverted the base class change

argagnej committed:
Complete tests for all CRUD operations on DNS resolver

argagnej committed:
Added tests for CRUD operations on inbound endpoint

argagnej committed:
Added tests for CRUD options on outbound endpoint

argagnej committed:
Added tests for CRUD operations on DNS forwarding ruleset

Yao725 committed:
empty generated folder

Yao725 committed:
regen

Yao725 committed:
address comments and fix tests

Yao725 committed:
Merge branch 'dev2/argagnej/addTrack2SdkChangesForDnsResolver' of https://github.com/arpit-gagneja/azure-sdk-for-net into dev2/argagnej/addTrack2SdkChangesForDnsResolver

Yao725 committed:
fix test

argagnej committed:
Added tests for CRUD options on forwarding rule

argagnej committed:
Merge branch 'dev2/argagnej/addTrack2SdkChangesForDnsResolver' of https://github.com/arpit-gagneja/azure-sdk-for-net into dev2/argagnej/addTrack2SdkChangesForDnsResolver

argagnej committed:
Fixed forwarding rule tests

argagnej committed:
Added tests for CRUD options on virtual network link

Yao725 commented on sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/Tests/DnsForwardingRulesetTests.cs:
```diff
@@ -0,0 +1,156 @@
+        }
+
+        [Test]
+        public async Task CreateDnsForwardingRuleset()
```
Can we add the attribute `RecordedTest` to all the test cases?

arpit-gagneja replied:
Update the attribute.

Yao725 commented on sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/DnsResolverTestBase.cs:
```diff
@@ -0,0 +1,99 @@
+            return rgOp.Value;
+        }
+
+        protected async Task CreateVirtualNetworkAsync(string virtualNetworkName)
```
Can we add a project reference to `Azure.ResourceManager.Network` and use it to create the vnet?

arpit-gagneja replied:
This is intentional. For recording, we have selected the ARM dogfood environment as we don't want to create so many resources in the prod environment. As of today, there is a limitation that virtual networks and subnets can't be created in the ARM dogfood. As a workaround, my team has written the simulator service to create VNETs and subnets. This function is calling that simulator service.

Yao725 commented on sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/DnsResolverTestBase.cs:
```diff
@@ -0,0 +1,99 @@
+            await client.PutAsync(baseUri + relativeUri, httpContent);
+        }
+
+        protected async Task CreateSubnetAsync(string virtualNetworkName)
```
Same as above

arpit-gagneja replied:
Replied to the above comment.

argagnej committed:
Changed the tests to recorded test attribute

Yao725 committed:
bring deleted file back

Yao725 committed:
update api

Yao725 committed:
prepare release
